### ISSUE TITLE
"Wrong date format" exception using sample code

### ISSUE THREAD

**Issue by maxmumford** [XeroAPI/Xero-Java#168]
**SDK you're using (please complete the following information):**
 - Version 3.1.6

**Describe the bug**
Following the setup instructions in this project's readme, when running this line to get a list of organisations:

`List<Connection> connection = idApi.getConnections(access_token);`

I get the following exception:

```
Exception in thread "main" com.fasterxml.jackson.databind.JsonMappingException: Wrong date format (through reference chain: java.util.ArrayList[0]->com.xero.models.identity.Connection["updatedDateUtc"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:394)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:353)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1711)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:290)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4014)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3078)
	at com.xero.api.client.IdentityApi.getConnections(IdentityApi.java:141)
	at xero.Callback.main(Callback.java:74)
Caused by: java.lang.IllegalArgumentException: Wrong date format
	at com.xero.api.CustomOffsetDateTimeDeserializer.deserialize(CustomOffsetDateTimeDeserializer.java:45)
	at com.xero.api.CustomOffsetDateTimeDeserializer.deserialize(CustomOffsetDateTimeDeserializer.java:18)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288)
	... 8 more
```

**To Reproduce**
Clone reproduction project: https://github.com/maxmumford/xero-wrong-date-format
In files Main.java and Callback.java, fill in the clientId and clientSecret variables.
In callback.java, give a placeholder value to "code" on line 38. We will fill this in later...
Open Main.java and set a breakpoint on line 58 - we're going to use the value from the url variable
Create a run configuration to run the Main.java file 
Run that run config in debug mode. When the breakpoint is hit, copy the value from the url variable
Visit the url in your web browser
Select your organisation
When you are redirected to localhost, copy the "code" value from the url query string
Enter this value into the Callback.java file in the "code" variable
Create a run configuration for the Callback.java file
Run it in debug mode
The exception will be thrown on line 78

**Expected behavior**
Callback.java:78 should not throw an exception and instead return a list of organisations

**Comment by sevendark** [comment]
updatedDateUtc proerty format in Connection returned by xero is "2019-12-27T05:47:17.0718110"

**Comment by Sayyiditow** [comment]
This error just started. It was not there past few days. Our application is also hitting this same error:

```
There was an unexpected error (type=Internal Server Error, status=500).
Wrong date format (through reference chain: java.util.ArrayList[0]->com.xero.models.identity.Connection["updatedDateUtc"])
```

**Comment by SidneyAllen** [comment]
@Sayyiditow @sevendark @maxmumford 

Thanks for flagging this and sorry about the issue.

I've confirmed the data parsing is breaking with the newly added date properties from the connections endpoint.

version 3.1.3 is the most recent version of the SDK that does not have this issue.  We will fix this when we return from the holidays.

**Comment by Sayyiditow** [comment]
Hi, Any news regarding this? Using 3.1.3 version is not good for us as we cannot get records using the if-modified-since parameter. - A bug I raised earlier.

**Comment by SidneyAllen** [comment]
@Sayyiditow @sevendark @maxmumford 

I've released 3.1.7 of Xero-Java SDK this addresses the date issue with  Connections endpoint from the identity API set.

**Comment by maxmumford** [comment]
Confirmed fixed in 3.1.7. Thanks for your help everybody 👍 Happy new year.

**Comment by Sayyiditow** [comment]
> Confirmed fixed in 3.1.7. Thanks for your help everybody  Happy new year.

I think this is 3.1.6? I could not find 3.1.7 on maven.

**Comment by maxmumford** [comment]
It's definitely there:

https://search.maven.org/artifact/com.github.xeroapi/xero-java/3.1.7/jar

Here's the dependency declaration I used:

```
        <dependency>
            <groupId>com.github.xeroapi</groupId>
            <artifactId>xero-java</artifactId>
            <version>3.1.7</version>
        </dependency>
```
